src/components/Drawer/Drawer.js

```diff
@@ -1,55 +1,112 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
+import { Transition } from 'react-transition-group';
 import bem from 'bem';
 import { IoIosArrowUp } from 'react-icons/io';
 import styles from './Drawer.scss';
 
+const TRANSITION_DURATION = {
+    enter: 10,
+    exit: 300,
+};
+
 const { block, elem } = bem({
     name: 'Drawer',
     classnames: styles,
-    propsToMods: ['isOpen'],
+    propsToMods: ['isShownAndExpanded', 'isShownAndClosed', 'isExpanded'],
 });
 
 const Drawer = props => {
-    const { title, initialIsOpen, children, ...rest } = props;
-    const [isOpen, setIsOpen] = useState(initialIsOpen);
+    const {
+        title,
+        isShown,
+        isExpanded: isExpandedProps,
+        initialIsExpanded,
+        onClick,
+        children,
+        ...rest
+    } = props;
+    const [isExpanded, setIsExpanded] = useState(isExpandedProps || initialIsExpanded);
+
+    useEffect(() => {
+        if (isExpandedProps !== null) {
+            setIsExpanded(isExpandedProps);
+        }
+    }, [isExpandedProps]);
+
+    function getStatus() {
+        const STATUS_LIST = {
+            IS_EXPANDED: 'isShownAndExpanded',
+            IS_CLOSED: 'isShownAndClosed',
+            IS_HIDDEN: '',
+        };
+
+        if (isShown) {
+            return isExpanded ? STATUS_LIST.IS_EXPANDED : STATUS_LIST.IS_CLOSED;
+        }
+
+        return STATUS_LIST.IS_HIDDEN;
+    }
 
     function handleClick() {
-        setIsOpen(!isOpen);
+        if (onClick) {
+            onClick();
+        } else {
+            setIsExpanded(!isExpanded);
+        }
     }
 
-    return (
-        <div {...block({ isOpen })} {...rest}>
-            <header {...elem('head', props)}>
-                <h4 {...elem('headTitle', props)}>{title}</h4>
-                <div {...elem('headButtons', props)}>
-                    <button
-                        {...elem('extendButton', { isOpen })}
-                        type="button"
-                        onClick={handleClick}
-                    >
-                        <IoIosArrowUp {...elem('extendIcon', props)} />
-                    </button>
+    return ReactDOM.createPortal(
+        <Transition appear unmountOnExit in={isShown} timeout={TRANSITION_DURATION}>
+            {state => (
+                <div
+                    {...block({ [getStatus()]: state !== 'entering' })}
+                    aria-hidden={!isShown}
+                    {...rest}
+                >
+                    <header {...elem('head', props)}>
+                        <h4 {...elem('headTitle', props)}>{title}</h4>
+                        <div {...elem('headButtons', props)}>
```

Comment:
is this div really needed? cannot we put the css all on the button itself?

Reply:
With div wrapper easy to add new buttons, now it is not the case, but in future it possible. We have in Search customer CSS, and often this CSS relies on HTML structure, so it is very painful to change HTML structure.

Reply:
I thought that the plan is not to allow custom css in the future but only change the theme variables we expose. Otherwise we will be very limited in many other places as well. Not important for this div, but for other places and general planning of component I would be concerned.